Cognite.Simulator.Utils/ConnectorBase.cs

```diff
@@ -212,50 +212,41 @@ protected async Task InitRemoteSimulatorIntegrations(CancellationToken token)
                     token).ConfigureAwait(false);
                 var integrations = integrationRes.Items;
                 var connectorName = GetConnectorName();
-                for (var index = 0; index < Simulators.Count; index++)
+                var simulatorExternalId = SimulatorDefinition.ExternalId;
+                    
+                var existing = integrations.FirstOrDefault(i => i.ExternalId == connectorName && i.SimulatorExternalId == simulatorExternalId);
+                if (existing == null)
                 {
-                    var simulator = Simulators[index];
+                    _logger.LogInformation("Creating new simulator integration for {Simulator}", SimulatorDefinition.Name);
+                    var existingSimulators = await Cdf.CogniteClient.Alpha.Simulators.ListAsync(
+                        new SimulatorQuery (),
+                        token).ConfigureAwait(false);
+                    var existingSimulator = existingSimulators.Items.FirstOrDefault(s => s.ExternalId == simulatorExternalId);
 
-                    // If there are more than one, we add the Simulator name as a prefix
-                    if (index > 0)
+                    if (existingSimulator == null)
                     {
-                        connectorName = $"{simulator.Name}-{GetConnectorName()}";
+                        _logger.LogWarning("Simulator {Simulator} not found in CDF", simulatorExternalId);
+                        throw new ConnectorException($"Simulator {simulatorExternalId} not found in CDF");
                     }
```

Comment:
do you think we need to do this here? because we are also checking and creating a simulator with the `GetOrCreateSimulator` in the `DefaultConnectorRuntime`

Reply:
true